verification/verifier.go

```diff
@@ -1,38 +1,159 @@
+		}
+		outcome := &SignatureVerificationOutcome{
+			VerificationResults: []*VerificationResult{},
+			VerificationLevel:   verificationLevel,
+		}
+		err = v.processSignature(sigBlob, sigManifest, trustPolicy, outcome)
+		if err != nil {
+			outcome.Error = err
+		}
+		verificationOutcomes = append(verificationOutcomes, outcome)
+	}
+
+	// check whether verification was successful or not
+	for _, outcome := range verificationOutcomes {
+
+		// all validations must pass
+		if outcome.Error != nil {
+			continue
+		}
+
+		// artifact digest must match the digest from the signature payload
+		payload := &notation.Payload{}
+		err := json.Unmarshal(outcome.SignerInfo.Payload, payload)
+		if err != nil || artifactDescriptor.Equal(payload.TargetArtifact) {
+			outcome.Error = fmt.Errorf("given digest %q does not match the digest %q present in the digital signature", artifactDigest, payload.TargetArtifact.Digest.String())
+			continue
+		}
+		outcome.SignedAnnotations = payload.TargetArtifact.Annotations
+
+		// signature verification succeeds if there is at least one good signature
+		return verificationOutcomes, nil
+	}
+
+	return verificationOutcomes, ErrorVerificationFailed{}
+}
+
+func (v *Verifier) processSignature(sigBlob []byte, sigManifest registry.SignatureManifest, trustPolicy *TrustPolicy, outcome *SignatureVerificationOutcome) error {
+	// verify integrity first. notation will always verify integrity no matter what the signing scheme is
+	signerInfo, result := v.verifyIntegrity(sigBlob, sigManifest, outcome)
+	outcome.SignerInfo = signerInfo
+	outcome.VerificationResults = append(outcome.VerificationResults, result)
+	if isCriticalFailure(result) {
+		return result.Error
+	}
+
+	// perform remaining validations based on the signing scheme
+	if err := v.defaultVerification(trustPolicy, outcome); err != nil {
+		return err
+	}
+
+	// no error
+	return nil
 }
 
-func (v *Verifier) Verify(artifactUri string) error {
-	/*
-		[DONE] Find the applicable trust policy, if none, return error
-		If signatureVerification is skip, then return without an error
-		Retrieve signature manifests
-		Return error if no signature manifests
-		For each signature manifest
-			Check the root cert hash is present in trust store hashes, otherwise fail early
-			Retrieve the signature envelope
-			Verify integrity
-				Signing cert produced the signature
-				Chain from signing cert to root cert is valid
-			Verify Authenticity
-				[DONE] Verify root of trust is established
-				[DONE] Verify trusted identites match from the policy
-			Verify expiry time of the signature is in the future
-			(NOT in RC1) Verify timestamping signature if present
-			(NOT in RC1) Verify revocation
-			Invoke plugin for extended verification
-	*/
-
-	// No error
+// defaultVerification performs verification for the default singing scheme `notary.default.x509`
```

Comment:
Why do we need `notary.default.x509`, as opposed to `notary.x509`
Shouldn't default be a configuration that might change in the future?

Reply:
@SteveLasker I agree that this PR should not implement any points that are proposed in https://github.com/notaryproject/notaryproject/pull/165 unless it is merged. I removed the comment.